stage #2.5/module #3 Spring Core/#5.1. Annotation-based configuration/content.md

```diff
@@ -0,0 +1,470 @@
+- [CommonAnnotationBeanPostProcessor][5]
+- [PersistenceAnnotationBeanPostProcessor][6]
+- [EventListenerMethodProcessor][7]
+
+> `<context:annotation-config/>` only looks for annotations on beans in the same application context in which it is
+> defined. This means that, if you put `<context:annotation-config/>` in a `WebApplicationContext` for a
+> `DispatcherServlet`, it only checks for `@Autowired` beans in your controllers, and not your services.
+
+## @Required
+
+The `@Required` annotation applies to bean property setter methods, as in the following example:
+
+```java
+public class MusicRatingService {
+
+    private MusicRatesFinder musicRatesFinder;
+
+    @Required
+    public void setMusicRatesFinder(MusicRatesFinder musicRatesFinder) {
+        this.musicRatesFinder = musicRatesFinder;
+    }
+
+    // ...
+}
+```
+
+This annotation indicates that the affected bean property must be populated at configuration time, through an explicit
+property value in a bean definition or through autowiring. The container throws an exception if the affected bean
+property has not been populated. This allows for eager and explicit failure, avoiding `NullPointerException` instances
+or the like later on. We still recommend that you put assertions into the bean class itself (for example, into an init
+method). Doing so enforces those required references and values even when you use the class outside of a container.
+
+The [RequiredAnnotationBeanPostProcessor][8] must be registered as a bean to enable support for the `@Required`
+annotation.
+
+
+> The `@Required` annotation and `RequiredAnnotationBeanPostProcessor` are formally deprecated as of
+> Spring Framework 5.1, in favor of using constructor injection for required settings (or a custom implementation of
+> `InitializingBean.afterPropertiesSet()` or a custom `@PostConstruct` method along with bean property setter methods).
+
+## @Autowired
+
+You can apply the `@Autowired` annotation to constructors, as the following example shows:
+
+```java
+public class MusicRatingService {
+
+    private final MusicRatesFinder musicRatesFinder;
+
+    @Autowired
+    public MusicRatingService(MusicRatesFinder musicRatesFinder) {
+        this.musicRatesFinder = musicRatesFinder;
+    }
+
+    // ...
+}
+```
+
+> As of Spring Framework 4.3, an `@Autowired` annotation on such a constructor is no longer necessary if the target bean
+> defines only one constructor to begin with. However, if several constructors are available and there is no
+> primary/default constructor, at least one of the constructors must be annotated with `@Autowired` in order to instruct
+> the container which one to use.
+
+You can also apply the `@Autowired` annotation to traditional setter methods, to methods with arbitrary names and
+multiple arguments, to fields and even mix it with constructors.
+
+You can also instruct Spring to provide all beans of a particular type from the `ApplicationContext` by adding the
```

Comment:
I believe we may remove content on lines 106-141, because it duplicates info from [#4. Dependency Injection/content.md] https://github.com/mjc-school/MJC-School/pull/72/files#diff-dcfae9cfd876fa96052a575f629aabc4a6c445e79b01e1e6077fb0b8f390ca69

Reply:
Content on lines 106-141 describes that Spring can provide beans into array or collection property, unfortunately I didn't find that in Dependency Injection section, am I missing something?

Reply:
OK, the main point here is providing beans into array or collection property, not just desribing `@Autowared` annotation. Lets leave it as is